eras/shelley/impl/src/Cardano/Ledger/Shelley/Genesis.hs

```diff
@@ -129,6 +146,52 @@ emptyGenesisStaking =
       sgsStake = mempty
     }
 
+-- | Unlike @'NominalDiffTime'@ that supports @'Pico'@ precision, this type
+-- only supports @'Micro'@ precision.
+newtype NominalDiffTimeMicro = NominalDiffTimeMicro NominalDiffTime
+  deriving (Show, Eq, Generic)
+  deriving anyclass (NoThunks)
+  deriving newtype (Ord, Num, Fractional, Real, ToJSON, FromJSON)
+
+instance ToCBOR NominalDiffTimeMicro where
+  toCBOR = encodeNominalDiffTimeMicro
+
+encodeNominalDiffTimeMicro :: NominalDiffTimeMicro -> Encoding
+encodeNominalDiffTimeMicro (NominalDiffTimeMicro ndtm) = encodeInteger . (`div` 1_000_000) . toPicoseconds $ ndtm
+  where
+    toPicoseconds t =
+      numerator (toRational t * toRational (resolution $ Proxy @E12))
+
+instance FromCBOR NominalDiffTimeMicro where
+  fromCBOR = decodeNominalDiffTimeMicro
+
+-- | For backwards compatibility we round pico precision to micro
+decodeNominalDiffTimeMicro :: Decoder s NominalDiffTimeMicro
+decodeNominalDiffTimeMicro = NominalDiffTimeMicro <$> fromRational . (% 1_000_000) <$> decodeInteger
+
+nominalDiffTimeMicroToSeconds :: NominalDiffTimeMicro -> Pico
+nominalDiffTimeMicroToSeconds (NominalDiffTimeMicro ndtm) = nominalDiffTimeToSeconds ndtm
+
+secondsToNominalDiffTimeMicro :: Pico -> NominalDiffTimeMicro
+secondsToNominalDiffTimeMicro (MkFixed s) =
+  NominalDiffTimeMicro $
+    secondsToNominalDiffTime $
+      MkFixed (1_000_000 * (s `div` 1_000_000))
+
+toNominalDiffTimeMicro :: NominalDiffTime -> Maybe NominalDiffTimeMicro
+toNominalDiffTimeMicro ndt
+  | ndtm == NominalDiffTimeMicro ndt = Just ndtm
+  | otherwise = Nothing
+  where
+    ndtm = toNominalDiffTimeMicroWithRounding ndt
+
+toNominalDiffTimeMicroWithRounding :: NominalDiffTime -> NominalDiffTimeMicro
+toNominalDiffTimeMicroWithRounding ndt =
+  secondsToNominalDiffTimeMicro $ nominalDiffTimeToSeconds ndt
+
+fromNominalDiffTimeMicro :: NominalDiffTimeMicro -> NominalDiffTime
+fromNominalDiffTimeMicro (NominalDiffTimeMicro ndt) = ndt
```

Comment:
This would become
```suggestion
fromNominalDiffTimeMicro (NominalDiffTimeMicro (MkFixed ndt)) = NominalDiffTimeMicro (MkFixed (ndtm * 1_000_000))
```

Reply:
I get it now! Thanks! :raised_hands: